core/indigo-core/common/math/algebra.h

```diff
@@ -822,5 +824,234 @@ namespace indigo
         float _d;
     };
 
+    inline bool isPointInPolygon(const Vec2f& p, const std::vector<Vec2f>& poly)
```

Comment:
I think we need more comments.
Also no need to calculate x if both vertices has x > px
So I suggest something like 

```c++
inline bool isPointInPolygon(const Vec2f& p, const std::vector<Vec2f>& poly)
{
    // Ray casting algorithm
    bool in = false;
    for (size_t i = 0, n = poly.size(); i < n; ++i)
    {
        size_t j = (i + 1) % n;
        if (((poly[i].y > p.y) != (poly[j].y > p.y)) && // point y between vertices and
            (((p.x > poly[i].x) && (p.x < poly[j].x)) || // both vertices are at right of point or edge is at right of point
            (p.x < (poly[i].x + (p.y - poly[i].y) * (poly[j].x - poly[i].x) / (poly[j].y - poly[i].y))))) // py < (py-y0)*dx/dy
            in = !in;
    }
    return in;
}
```